testsuite/model/src/test/java/org/keycloak/testsuite/model/ClientModelTest.java

```diff
@@ -76,6 +90,47 @@ public void testClientsBasics() {
             assertThat(model.getClientId(), is(equalTo("myClientId")));
         }
 
+        // Search by clientId
+        {
+            AtomicReference<String> id = new AtomicReference<>();
+            AtomicReference<String> clientId = new AtomicReference<>();
+
+            withRealm(realmId, (session, realm) -> {
+                getClientIds(session.clients().searchClientsByClientIdStream(realm, "client with", 0, 10), id, clientId);
+                return null;
+            });
+            assertThat(id.get(), is(equalTo(searchClient.getId())));
+            assertThat(clientId.get(), is(equalTo(searchClientId)));
+
+            withRealm(realmId, (session, realm) -> {
+                getClientIds(session.clients().searchClientsByClientIdStream(realm, "sp*ci_l Ch***cters", 0, 10), id, clientId);
+                return null;
+            });
+            assertThat(id.get(), is(equalTo(searchClient.getId())));
```

Comment:
Just a nitpick: it would be easier to move `assertThat` into the lambda in the `withRealm` call. This way you would avoid setting `id` and `clientId` to the `AtomicRefence`.